src/server/game/Spells/SpellMgr.cpp

```diff
@@ -81,6 +81,9 @@ DiminishingGroup GetDiminishingReturnsGroupForSpell(SpellInfo const* spellproto,
                 // Screams of the Past
                 else if (spellproto->Id == 7074)
                     return DIMINISHING_NONE;
+                // Swoop
+                else if (spellproto->Id == 37012)
+                    return DIMINISHING_NONE;
```

Comment:
minor: if you have a `return` inside a `if / else if` you could convert the `else if` into `if`.

```C++
                // Pet charge effects (Infernal Awakening, Demon Charge)
                if (spellproto->SpellVisual[0] == 2816 && spellproto->SpellIconID == 15)
                    return DIMINISHING_CONTROLLED_STUN;
                // Gnaw
                if (spellproto->Id == 47481)
                    return DIMINISHING_CONTROLLED_STUN;
                // Screams of the Past
                if (spellproto->Id == 7074)
                    return DIMINISHING_NONE;
                // Swoop
                if (spellproto->Id == 37012)
                    return DIMINISHING_NONE;
```

Reply:
I know it. Just don;t want to break codestyle there.

Reply:
ok, no worries at all and thanks for the fix ;-)